cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -523,6 +525,17 @@ ReconciliationState createReconciliationState(Reconciliation reconciliation, Kaf
             return initialStatusPromise.future();
         }
 
+        /**
+         * Checks the requested Kafka spec for potential issues, and adds warnings and advice for best
+         * practice to the status.
+         */
+        Future<ReconciliationState> checkKafkaSpec() {
+            KafkaSpecChecker checker = new KafkaSpecChecker(kafkaAssembly.getSpec(), kafkaCluster, zkCluster);
+            List<Condition> warnings = checker.run();
```

Comment:
I wonder if we should also add warnings from the `ValidationVisitor`. Potentially that could add quite a few warnings, but it makes them more visible than just in the logs. WDYT @scholzj ?

Reply:
Sorry, I missed this earlier. But I think this would be good idea for next PR.